airgun/entities/filter.py

```diff
@@ -29,6 +29,24 @@ def read(self, role_name, entity_name, widget_names=None):
             self, 'Edit', role_name=role_name, entity_name=entity_name)
         return view.read(widget_names=widget_names)
 
+    def read_all(self, role_name):
+        """Read all the available role filters table values"""
+        view = self.navigate_to(self, 'All', role_name=role_name)
+        return view.table.read()
+
+    def read_permissions(self, role_name):
+        """Return all role filters permissions as a dict with resources as keys and permission
+        names as values.
+        """
+        rows = self.read_all(role_name)
+        permissions = {}
+        for row in rows:
+            resource = row['Resource']
+            if resource not in permissions:
+                permissions[resource] = []
+            permissions[resource].extend(row['Permissions'].split(', '))
```

Comment:
You can drop `if` and use this instead: 
```
permissions.setdefault(resource, []).extend(row['Permissions'].split(', '))
```

Reply:
Nope we need to return a pure dict, (that was my first implementation :) )

Reply:
Nope my first implementation was with permissions = defaultdict([]), any way do not think this can be a blocker as it's really complicate the reading of this line.

Reply:
if everybody vote for this will implement it

Reply:
I don't usually encourage redundant conditions when there's shorter version, but i have to agree in this case option with `if` is better readable, 3 func calls in a single line is too much and it's hard to understand what's going on from the first glance.

Reply:
it is not so important